### PR TITLE
[SPARK-9155][SQL] codegen StringSpace

### PR DESCRIPTION
**Pull request by tarekbecker**
Jira https://issues.apache.org/jira/browse/SPARK-9155

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringOperations.scala [hunk 1]
```diff
@@ -556,6 +556,16 @@ case class StringSpace(child: Expression)
     UTF8String.fromBytes(spaces)
   }
 
+  override def genCode(ctx: CodeGenContext, ev: GeneratedExpressionCode): String = {
+    nullSafeCodeGen(ctx, ev, (length) => {
+      val spaces = ctx.freshName("spaces")
+      s"""
+        byte[] $spaces = new byte[($length < 0) ? 0 : $length];
```
Comment (rxin) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringOperations.scala]: can we move this into UTF8String, and just call that function both in the interpreted mode and in codegen?

Comment (tarekauel) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringOperations.scala]: Okay

**Comment by tarekauel** [comment]
Jenkins, test this please

File: unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java [hunk 2]
```diff
@@ -77,6 +78,15 @@ public static UTF8String fromString(String str) {
     }
   }
 
+  /**
+   * Creates an UTF8String that contains `length` spaces.
+   */
+  public static UTF8String blankString(int length) {
+    byte[] spaces = new byte[length];
+    Arrays.fill(spaces, (byte) ' ');
```
Comment (rxin) [inline unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java]: does ' ' always become space if utf8 encoding is not the default in the system? might be better to explicitly write down 0x20

**Comment by rxin** [comment]
Jenkins, ok to test.

File: unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java (hunk 2 above)
Comment (tarekauel) [inline unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java]: Char implements UTF-16 (http://docs.oracle.com/javase/7/docs/api/java/lang/Character.html)

"The char data type (and therefore the value that a Character object encapsulates) are based on the original Unicode specification, which defined characters as fixed-width 16-bit entities".

Shall I still change it?

Comment (rxin) [inline unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java]: ok in that case it is fine

**Pull request closed without merging**